packages/compiler-cli/ngcc/src/writing/cleaning/cleaning_strategies.ts

```diff
@@ -0,0 +1,62 @@
+/**
+ * @license
+ * Copyright Google Inc. All Rights Reserved.
+ *
+ * Use of this source code is governed by an MIT-style license that can be
+ * found in the LICENSE file at https://angular.io/license
+ */
+import {AbsoluteFsPath, FileSystem, PathSegment, absoluteFrom} from '../../../../src/ngtsc/file_system';
+import {cleanPackageJson} from '../../packages/build_marker';
+import {NGCC_BACKUP_EXTENSION} from '../in_place_file_writer';
+import {NGCC_DIRECTORY} from '../new_entry_point_file_writer';
+import {isLocalDirectory} from './utils';
+
+/**
+* Implement this interface to extend the cleaning strategies of the `PackageCleaner`.
+*/
+export interface CleaningStrategy {
+  canClean(path: AbsoluteFsPath, basename: PathSegment): boolean;
+  clean(path: AbsoluteFsPath, basename: PathSegment): void;
+}
+
+/**
+ * A CleaningStrategy that reverts changes to package.json files by removing the build marker and
+ * other properties.
+ */
+export class PackageJsonCleaner implements CleaningStrategy {
+  constructor(private fs: FileSystem) {}
+  canClean(_path: AbsoluteFsPath, basename: PathSegment): boolean {
+    return basename === 'package.json';
+  }
+  clean(path: AbsoluteFsPath, _basename: PathSegment): void {
+    const packageJson = JSON.parse(this.fs.readFile(path));
+    cleanPackageJson(packageJson);
+    this.fs.writeFile(path, JSON.stringify(packageJson));
```

Comment:
Do we do this already? `JSON.stringify` will return a one-line blob with no whitespace. It's not a good experience for users if ngcc compresses their package.json like that.

I think we should at _least_ do 
```suggestion
    this.fs.writeFile(path, JSON.stringify(packageJson, null, 2));
```
which will apply some prettier formatting. But I'd prefer if we could avoid reformatting any parts of the file besides the keys we touch.

Reply:
Good spot. Currently we do this:

```
      this.fs.writeFile(path, `${JSON.stringify(packageJson, null, 2)}\n`);
```

I will update this bit of code to follow this pattern - but also not to write the file at all if we didn't change it.

Reply:
That's what the `PackageJsonUpdater` does (`JSON.stringify(..., null, 2)` + `'\n'`). I think npm/yarn may do the same, so I believe it is fine in this case.